probers/dns/api.go

```diff
@@ -0,0 +1,27 @@
+package dns
+
+import (
+	"github.com/Symantec/tricorder/go/tricorder"
+)
+
+type dnsconfig struct {
+	testname string
+	hostname string
+	healthy  bool
+	latency  int64
+}
+
+func (p *dnsconfig) Register(dir *tricorder.DirectorySpec) error {
+	return p.register(dir)
+}
+
+func (p *dnsconfig) Probe() error {
+	return p.probe()
+}
+
+func Makednsprober(testname, hostname string) *dnsconfig {
+	p := new(dnsconfig)
+	p.testname = testname
+	p.hostname = hostname
```

Comment:
An alternative method would be:
p := &dnsconfig{testname: testname, hostname: hostname}

Your choice. I find that this form is generally cleaner over the long haul.